core/org.wso2.carbon.user.core/src/main/java/org/wso2/carbon/user/core/UniqueIDUserStoreManager.java

```diff
@@ -447,36 +468,36 @@ void updateUserListOfRoleWithID(String roleName, String[] deletedUserIDs, String
      * Add a group to the system.
      *
      * @param groupName   Group's display name.
-     * @param userIDs     List of User IDs belongs to the group.
+     * @param users       List of User IDs belongs to the group.
      * @param permissions List of permissions of the group.
-     * @param attributes  Group attributes values.
+     * @param claims      List of Claims.
      * @return created Group object.
      * @throws UserStoreException If an error occurs while adding a group.
      */
-    Group addGroup(String groupName, List<String> userIDs, List<Permission> permissions,
-                   Map<String, String> attributes) throws UserStoreException;
+    Group addGroup(String groupName, List<User> users, List<Permission> permissions,
+                   List<Claim> claims) throws UserStoreException;
 
     /**
      * Update group claim values.
      *
      * @param groupID     Group ID.
-     * @param attributes  Map of attribute values. These values will be replaced.
+     * @param claims      List of claims. These values will be replaced.
      * @param permissions List of permissions.
      * @return Updated group.
      * @throws UserStoreException If an error occurs while updating group.
      */
-    Group updateGroup(String groupID, Map<String, String> attributes,
+    Group updateGroup(String groupID, List<Claim> claims,
```

Comment:
This claim object contains claim metadata. Do you need claim metadata as parameters?

Reply:
In the future we might be able to use. If we need to add any extra info in the future, no need of an API change. We can directly bind to the claim object

Reply:
I don't see any reason to parse claim metadata as parameters in the update group request. If needed we can internally retrieve the claim information using the claim API.

Reply:
created new claim in the common package